### ISSUE TITLE
postcss-easy-import

### ISSUE THREAD

**Issue by TrySound** [postcss/postcss-import#145]
# postcss-easy-import

What if we will create pack around `postcss-import` with many optional things like `prefix`, `extensions`, `glob` etc.

How it could be named?

### prefix

Type: `String`
Default: `_`

Add before imported filename.

`@import "dir/partial"` will look for `dir/_partial.css` with `prefix: '_'`

### extensions

Type: `Array` or `String`
Default: `.css`

### glob

Type: `Boolean`
Default: `false`

**Comment by ben-eb** [comment]
`postcss-import-partial`
`postcss-import-glob`
`postcss-import-extensions`

?

Idea is solid. :+1:

**Comment by TrySound** [comment]
@ben-eb No, no. I mean plugin with a few new options which will be wrap postcss-import

```
postcssPackImport({
  partial: false,
  extensions: ['.css'],
  glob: true
})
```

**Comment by MoOx** [comment]
Sounds weird and will create confusion imo.

**Comment by TrySound** [comment]
@MoOx postcss-import can be strict and spec followed but with extending ability. postcss-pack-import (or another name) will use postcss-import as a dependency and realise scss-like features without any complications. Just for lazy people.

**Comment by TrySound** [comment]
Modularity celebration :)

**Comment by MoOx** [comment]
I get your point, but most people are lazy.
Find a good name first, then we will talk :D.
Kidding.
What about postcss-easy-import ?

**Comment by TrySound** [comment]
Make sense.

/cc @jonathantneal

**Comment by TrySound** [comment]
/cc @MoOx @jonathantneal 
Done https://github.com/TrySound/postcss-easy-import

**Comment by MoOx** [comment]
Nice, can you make a PR to README and add a comment/link in 8.0 CHANGELOG section so we can close this issue?
Thanks a ton!

**Comment by jonathantneal** [comment]
So this is a wrapper to give postcss-import the functionality of postcss-partial-import? Honest question, if we could just refactor postcss-import, would the functionality belong in a separate plugin?

**Comment by TrySound** [comment]
postcss-import is simple and powerful api with "all we need" defaults. easy-import just gives extra very specific functionality. Powerful of modularity :)

**Comment by jonathantneal** [comment]
Why should this sugar be in its own repo? I don’t like there being 3 repos now.

For instance, postcss-import does sluething to resolve the path of an `@import` rule. It will look in process.cwd(), web_modules, node_modules and local modules, right? Is that not sugar? Is it part of a spec?

**Comment by TrySound** [comment]
@jonathantneal You are right. But it's simple resolver and common usable stuff. And easy-import includes all postcss-import features like media queries and plugins option for linting

**Comment by TrySound** [comment]
@jonathantneal `postcss-import` and `postcss-partial-import` are not competitors. Your module is inliner syntax "sugar". It can inline files inside rules or at-rules. But you can't realize spec media resolver. I did it in postcss-import. It has constraints. But it works like bundler. postcss-easy-import is just spec + some resolve help for sass lovers.

**Comment by MoOx** [comment]
This thing is done, so closing.

**Comment by simonsmith** [comment]
Could we consider bringing some of this functionality back in? `easy-import` has been abandoned from what I can see and useful changes [like this one](https://github.com/TrySound/postcss-easy-import/pull/7) have been left unmerged.

I think globbing is a very useful and popular addition.

cc @RyanZim

**Comment by RyanZim** [comment]
@simonsmith First, this won't happen in v9.0.0; it's long overdue for release. Hope that gets out in the next few days.

Second, this is highly bikeshedable. Perhaps you can open a new issue outlining some of your thoughts; when consensus is reached, we can proceed to implement.

**Comment by TrySound** [comment]
Who want to get access to this?

**Comment by RyanZim** [comment]
@TrySound I hardly have time; @simonsmith, you feel like trying?

**Comment by simonsmith** [comment]
If @TrySound is happy to let others contribute to `postcss-easy-import` then maybe we can just continue the work on that repo. 

I'd be happy to help merge the outstanding PR and have a look at the remaining issues.

**Comment by RyanZim** [comment]
@simonsmith Go for it!

**Comment by simonsmith** [comment]
For reference just published `2.0.0` which fixed all outstanding issues and uses latest `postcss-import` - https://github.com/TrySound/postcss-easy-import/releases/tag/2.0.0
